app/models/tagged_animal_assessment.rb

```diff
@@ -69,8 +69,31 @@ def self.create_from_csv_data(attrs)
     end
   end
 
-  def self.lengths_for_measurement(processed_file_id)
-    select(:length).where(processed_file_id: processed_file_id).map { |record| record.length.to_f }
+  def self.lengths_for_measurement(shl_case_number, measurement_date)
+      measurements = select(:length)
+                     .where(shl_case_number: shl_case_number)
```

Comment:
Super nit picky, but you can actually combine these two `where` clauses into one, not sure which way it reads better though but thought I should just mention :)